src/modules/content-s3/content-s3.c

```diff
@@ -79,67 +127,35 @@ static int parse_config (struct content_s3 *ctx,
                           "uri",
                           &uri,
                           "virtual-host-style",
-                          &ctx->is_virtual_host) < 0) {
+                          &is_virtual_host) < 0) {
         snprintf(errbuff, eb_size, "failed to parse config file");
         goto error;
     }
 
     if (!(cpy = strdup (uri)))
         goto error;
 
-    if (yuarel_parse (&yuri, (char *) cpy) < 0) {
+    if (yuarel_parse (&yuri, cpy) < 0) {
         snprintf(errbuff, eb_size, "failed to parse uri");
         goto error;
     }
-
-    ctx->cfg->hostname = yuri.host;
-    if (!(strncat (ctx->cfg->hostname, ":", sizeof(ctx->cfg->hostname) + 1))) {
-        snprintf(errbuff, eb_size, "failed to form hostname");
-        goto error;
-    }
-
+    
     sprintf(tmp, "%d", yuri.port);
-    if (!(strncat (ctx->cfg->hostname, tmp, sizeof(ctx->cfg->hostname) + sizeof(yuri.port)))) {
+    if (!(cfg->hostname = hostport (yuri.host, tmp))) {
```

Comment:
Could just make this function take an integer port.  (Sorry when I suggested it earlier I was thinking the yuri.port was a string).  Looks like yuarel sets the port to zero if it's not set in the input, so hostport would just test for 0 instead of NULL for omitting :port from the output string.